app/src/main/java/com/internship/move/feature/map/MapFragment.kt

```diff
@@ -25,10 +33,9 @@ class MapFragment : Fragment(R.layout.fragment_map) {
         }
 
         binding.clearBtn.setOnClickListener {
-            requireActivity().getPreferences(Context.MODE_PRIVATE)
-                .edit()
-                .putBoolean("IS_LOGGED", false)
-                .apply()
+            scope.launch {
```

Comment:
better use `viewModelScope.launch`. The view might get destroyed faster and the scope cancelled.
Move it in the viewmodel